MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -214,23 +214,32 @@ - (void)requestInstallInformationWith:(NSString *)releaseHash {
     if (url) {
 
 /*
- * Only iOS 9.x and 10.x will download the update after users click the "Install" button. 
+ * Only iOS 9.x and 10.x will download the update after users click the "Install" button.
  * We need to force-exit the application for other versions or for any versions when the update is mandatory.
  */
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wpartial-availability"
       Class clazz = [SFSafariViewController class];
-#pragma clang diagnostic pop
-      /*
-       * TODO Checking operating system version is a workaround for iOS 11 where SFSafariViewController can't read Safari's cookies.
-       * Revert this change when SFAuthenticationSession will be ready.
-       */
       if (clazz && ![NSProcessInfo.processInfo isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){11, 0, 0}]) {
```

Comment:
I think there is two possibility to get rid of the heavy version check logic introduced in the workaround.
Either use feature availability. Something like
```
if SFAuthenticationSession class exists
  ...
else if SFSafariViewController class exists
  ...
else
  ...
```
Or the new `@available`.